### PR TITLE
Pluggable parsers for step definitions.

### LINKED ISSUE #91: Simpler text matcher idea
Regex is powerful tool but sometimes this power is not needed.
Probably for most cases just simple parser could be used (idea stolen from behave framework) – https://pypi.python.org/pypi/parse could be used as supplement to regexes in case of matching scenarios.

In that case step definition could be like:
```
@Given("I have {number} items")
@Given("the temperature is {temperature:float}C")
```

### PR DESCRIPTION
**Pull request by bubenkoff**
closes #91

### CODE REVIEW

File: pytest_bdd/parsers.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_cfparse.Parser(self.name, **kwargs)
+
+
+class string(StepArgumentParser):
+
+    """exact string step parser."""
```
Comment (olegpidsadnyi) [inline pytest_bdd/parsers.py]: Exact

Comment (bubenkoff) [inline pytest_bdd/parsers.py]: done

On 7 January 2015 at 15:26, Oleg Pidsadnyi <notifications@github.com> wrote:

> In pytest_bdd/parsers.py
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96#discussion-diff-22587927>
> :
>
> > +            return False
> > +
> > +
> > +class cfparse(parse):
> > +
> > +    """cfparse step parser."""
> > +
> > +    def __init__(self, name, **kwargs):
> > +        """Compile parse expression."""
> > +        super(parse, self).__init__(name)
> > +        self.parser = base_cfparse.Parser(self.name, **kwargs)
> > +
> > +
> > +class string(StepArgumentParser):
> > +
> > +    """exact string step parser."""
>
> Exact
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96/files#r22587927>.
>



-- 
Anatoly Bubenkov

File: pytest_bdd/parsers.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    """cfparse step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_cfparse.Parser(self.name, **kwargs)
+
+
+class string(StepArgumentParser):
```
Comment (olegpidsadnyi) [inline pytest_bdd/parsers.py]: Still it is a bit strange. That ArgumentParser doesn't parse any arguments. This is more like exact marcher indeed. Maybe rebrand it to Pattern?

File: pytest_bdd/parsers.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+class parse(StepArgumentParser):
+
+    """parse step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_parse.compile(self.name, **kwargs)
+
+    def get_params(self, name):
```
Comment (olegpidsadnyi) [inline pytest_bdd/parsers.py]: parse_params?

File: pytest_bdd/parsers.py (hunk 2 above)
Comment (bubenkoff) [inline pytest_bdd/parsers.py]: will rename to StepParser

On 7 January 2015 at 15:29, Oleg Pidsadnyi <notifications@github.com> wrote:

> In pytest_bdd/parsers.py
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96#discussion-diff-22588103>
> :
>
> > +            return bool(self.parser.parse(name))
> > +        except ValueError:
> > +            return False
> > +
> > +
> > +class cfparse(parse):
> > +
> > +    """cfparse step parser."""
> > +
> > +    def __init__(self, name, **kwargs):
> > +        """Compile parse expression."""
> > +        super(parse, self).__init__(name)
> > +        self.parser = base_cfparse.Parser(self.name, **kwargs)
> > +
> > +
> > +class string(StepArgumentParser):
>
> Still it is a bit strange. That ArgumentParser doesn't parse any
> arguments. This is more like exact marcher indeed. Maybe rebrand it to
> Pattern?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96/files#r22588103>.
>



-- 
Anatoly Bubenkov

File: pytest_bdd/parsers.py (hunk 3 above)
Comment (bubenkoff) [inline pytest_bdd/parsers.py]: parse_arguments, to be precise

On 7 January 2015 at 15:30, Oleg Pidsadnyi <notifications@github.com> wrote:

> In pytest_bdd/parsers.py
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96#discussion-diff-22588184>
> :
>
> > +
> > +    def is_matching(self, name):
> > +        """Match given name with the step name."""
> > +        return bool(self.regex.match(name))
> > +
> > +
> > +class parse(StepArgumentParser):
> > +
> > +    """parse step parser."""
> > +
> > +    def __init__(self, name, **kwargs):
> > +        """Compile parse expression."""
> > +        super(parse, self).__init__(name)
> > +        self.parser = base_parse.compile(self.name, **kwargs)
> > +
> > +    def get_params(self, name):
>
> parse_params?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96/files#r22588184>.
>



-- 
Anatoly Bubenkov

**Pull request merged**